### DISCUSSION TITLE
How to move two repositories in a new one?

### DISCUSSION THREAD

**Discussion by DianVK** [community/community#41444] [New to GitHub]
Hello, I need help, what I am trying to do is move 2 repositorys inside a new one, as shown on the picture.Could you help me ?https://prikachi.net/image/8wBUpU

**Comment by airtower-luna** [comment]
What exactly do you mean by moving the two repositories into one?

**Comment by DianVK** [comment]
Hello,
https://prikachi.net/image/8wDFU2 this is what i want to make,
Repository 1
Repository 2
move them into Repository 3 and keep all the history of commits
But with folders names not just the files which are in the repository
so what i want to happen is

Repository 3 > Repository1[folder] > files from repository1

 
 
 
 
 
 
 
 
 
 
 > Repository2[folder] > files from repository2
is this possible? Because I have been trying to make this from 2 days straight , I even made attempt to pay people from fiverr to make it for me , but they couldnt succeed too.
Is there any possible solutions for it or it is not possible because of conflict?
 When we try to make it , it is even saving only one of the repository history or not even from one of them when we put them in a folder of the new repository.









 >-------- Оригинално писмо --------

 >От: Airtower ***@***.***

 >Относно: Re: [community/community] How to move two repositories in a new one?
 (Discussion #41444)

 >До: community/community  

 >Изпратено на: 11.12.2022 22:59



 
 
  
  
  
What exactly do you mean by moving the two repositories into one?

**Comment by airtower-luna** [comment]
It's possible, if a little messy:

1. In each of the old repositories, move the files so they are in a subdirectory as you want them to be in the new repository. Make sure there's no overlap (common paths) between them or the new repository or you'll get merge conflicts.
2. In a local copy of the new repository, add the two old repositories as remotes and fetch them.
3. Merge the right branch of each of the old repositories. You'll need to use [`--allow-unrelated-histories`](https://git-scm.com/docs/git-merge#Documentation/git-merge.txt---allow-unrelated-histories).
4. Double check if everything is as you want, and if yes, push.

**Reply by Zackaryjacob** [reply]
https://developercertificate.org/
